### PR TITLE
Combine recognize and apply redirects

### PR DESCRIPTION
**Pull request by atscott**

### CODE REVIEW

File: goldens/size-tracking/integration-payloads.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   "cli-hello-world-lazy": {
     "uncompressed": {
       "runtime": 2734,
-      "main": 229111,
+      "main": 225424,
```
Comment (AndrewKushnir) [inline goldens/size-tracking/integration-payloads.json:29]: 🎉

File: packages/router/src/recognize.ts [hunk 2]
```diff
@@ -7,77 +7,107 @@
...
-          if (children === null) {
-            return null;
-          }
+        .pipe(
+            catchError((e: any) => {
+              if (e instanceof AbsoluteRedirect) {
+                // After an absolute redirect we do not apply any more redirects!
+                // If this implementation changes, update the documentation note in `redirectTo`.
+                this.allowRedirects = false;
+                // TODO: there were no tests covering this line locally. Only g3 failures
```
Comment (AndrewKushnir) [inline packages/router/src/recognize.ts:68]: Is it something we'd need to address in this PR?

**Review by AndrewKushnir — COMMENTED** [review]
@atscott I've looked through the change and I'd like to ask if we can pair to view it together, so I can better understand the context. Thank you.

File: packages/router/src/recognize.ts (hunk 2 above)
Comment (atscott) [reply packages/router/src/recognize.ts:68]: I think this was probaby meant for a different spot. There _are_ tests which cover this line.

**Review by AndrewKushnir — APPROVED** [review]
👍

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: size-tracking

**Comment by atscott** [comment]
Green TGP: https://fusion2.corp.google.com/presubmit/tap/519850236/OCL:519850236:BASE:520045700:1680019909910:3411ac52/targets

**Comment by atscott** [comment]
This PR was merged into the repository by commit 1600687fe518e67adcc629c78857720a5118d489.

**Comment by atscott** [comment]
Re-merging. I've confirmed this was an issue with the internal route configuration.

**Comment by atscott** [comment]
This PR was merged into the repository by commit 3c7e63737407287986c65136efd1f53d1215a53e.

**Pull request closed without merging**